core/client/db/src/lib.rs

```diff
@@ -265,21 +271,46 @@ impl<'a> state_db::MetaDb for StateMetaDb<'a> {
 	}
 }
 
+struct HeaderCache<Block: BlockT> {
+	hash_to_data: LruCache<Block::Hash, LightHeader<Block>>,
```

Comment:
architecturally, I don't follow why this would have to be in the `db` layer, when it is specifically a cache above the DB

Reply:
I guess I can move the methods in `Header` trait a bit up to `client::blockchain::Backend`, where you have similar things like `leaves,() children(), cache()`. And still be able to use it from `is_descendent_of` and `tree_route`.

But I was trying to maximize usage, so people don't read full headers from the DB so often.

I agree that from the architectural viewpoint is not so nice.

Reply:
Suggestion:
  1. introduce a separate trait for `HeaderMetadata`.
  2. implement it in a separate crate
  3. use this crate in `client-db` and do trait implementations + forwarding trait implementations for client, backends, etc.

Reply:
It seems like a good idea, but what would be the scope of such crate? what would be the methods to implement in the crate for `HeaderMetadata`? I guess more than just `set/get`, otherwise is too much overhead.

Reply:
i don't understand what you mean. it's completely fine to have a trait with only 1 or 2 methods, and to extract that out to a separate crate.